frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx

```diff
@@ -28,92 +64,165 @@ export const ValutConfigure: React.FC<KMSConfigureProps> = ({
 }) => {
   const { t } = useTranslation();
 
-  const kms: VaultConfig = state.kms?.[ProviderNames.VAULT] || state.kms;
-  const kmsObj: VaultConfig = _.cloneDeep(kms);
+  // webhooks
+  const isKmsVaultSASupported = useFlag(GUARDED_FEATURES.OCS_KMS_VAULT_SERVICE_ACCOUNT);
+
+  const vaultState: VaultConfig = state.kms?.[ProviderNames.VAULT] || state.kms;
+  const vaultStateClone: VaultConfig = _.cloneDeep(vaultState);
+  const { encryption } = state;
+
+  const updateVaultState = (vaultConfig: VaultConfig) => {
+    mode
+      ? setEncryptionDispatch(ActionType.SET_KMS_ENCRYPTION, mode, dispatch, vaultConfig)
+      : dispatch({
+          type: 'securityAndNetwork/setVault',
+          payload: vaultConfig,
+        });
+  };
+
+  const setAuthMethod = (authMethod: VaultAuthMethods) => {
+    vaultStateClone.authMethod = authMethod;
+    updateVaultState(vaultStateClone);
+  };
+
+  const filteredVaultAuthMethodMapping = Object.values(VaultAuthMethodMapping).filter(
+    (authMethod) =>
+      (encryption.clusterWide
+        ? authMethod.supportedEncryptionType.includes(KmsEncryptionLevel.CLUSTER_WIDE)
+        : true) &&
+      (encryption.storageClass
+        ? authMethod.supportedEncryptionType.includes(KmsEncryptionLevel.STORAGE_CLASS)
+        : true),
+  );
+
+  // set token as default auth method
+  const vaultAuthMethods = filteredVaultAuthMethodMapping.map((authMethod) => authMethod.value);
+  if (!vaultAuthMethods.includes(vaultState.authMethod)) {
+    setAuthMethod(VaultAuthMethods.TOKEN);
+  }
 
-  const [revealToken, setRevealToken] = React.useState(false);
+  return (
+    <>
+      {isKmsVaultSASupported && (
+        <FormGroup
+          fieldId="authentication-method"
+          label={t('ceph-storage-plugin~Authentication Method')}
+          className={`${className}__form-body`}
+          helperTextInvalid={t('ceph-storage-plugin~This is a required field')}
+          isRequired
+        >
+          <FormSelect
+            value={vaultState.authMethod}
+            onChange={setAuthMethod}
+            id="authentication-method"
+            name="authentication-method"
+            aria-label={t('ceph-storage-plugin~authentication-method')}
+            isDisabled={isLengthUnity(vaultAuthMethods)}
```

Comment:
Shouldn't this be  `isDisabled={isLengthUnity(vaultAuthMethods) || !isWizardFlow}` ??
We want to disable the dropdown when we are using StorageClass flow.

Reply:
Yes, either i need use isDisabled={isLengthUnity(vaultAuthMethods) || !isWizardFlow} (or) i have to pass 
entityType{storageClass: true, clusterEncryptionWide: false} from ocs-storage-class-form.tsx

So ```VaultAuthMethodMapping``` will takecare of it:

Reply:
I thought its ok to display this option on the storage-class screen. because only token-based auth is displayed. But you its ok to disable also.

will update code with this check.

Reply:
I feel it's ok to display this field on the storage class screen, anyway we will need this in the future.
 
![image](https://user-images.githubusercontent.com/15073590/144022534-edb90fdd-c674-4205-be89-d1617e487cc2.png)